### PR TITLE
NXP-25659: Add option to block modifier keys.

### PR DESCRIPTION
**Pull request by nmpcunha**

### CODE REVIEW

**Comment by nmpcunha** [comment]
For now, a flag was added to block the usage of modifier keys.

Anyway, since there is already a ticket to review our accessibility shortcuts (https://jira.nuxeo.com/browse/DESIGN-250), it would be interesting to think further about this, since there are some limitations from iron-a11y-keys that won't allow us to specify shortcuts with and without modifier keys: e.g. for one action having both shortcuts _ctrl+a_ or _v_ only

File: elements/nuxeo-keys/nuxeo-keys.html [hunk 1]
```diff
@@ -102,6 +93,39 @@
...
+          res.push(bind[0]);
+          return res;
+        }, []).filter(function(bind) {
+          return bind.key.toLowerCase() === this._transformKey(keyboardEvent.key) && (keyboardEvent.altKey
+              || keyboardEvent.ctrlKey || keyboardEvent.metaKey || keyboardEvent.shiftKey) === !!bind.hasModifiers;
+        }.bind(this));
+      },
+
+      /**
+       * Transforms the key to match iron-a11y-keys logic.
```
Comment (nmpcunha) [inline elements/nuxeo-keys/nuxeo-keys.html]: This is copied (and simplified) function from iron-a11y-keys-behavior

**Comment by nmpcunha** [comment]
The two different commits represent two different approaches we can take.

File: elements/nuxeo-keys/nuxeo-keys.html [hunk 2]
```diff
@@ -77,6 +93,40 @@
...
         }
         this.fire('pressed', e.detail, {});
+      },
+
+      _getMatchingKeyBindings: function(keyboardEvent) {
+        return this.$.a11y._keyBindings[keyboardEvent.type].reduce(function(res, bind) {
+          res.push(bind[0]);
+          return res;
+        }, []).filter(function(bind) {
+          return bind.key.toLowerCase() === this._transformKey(keyboardEvent.key)
```
Comment (Gabez0r) [inline elements/nuxeo-keys/nuxeo-keys.html]: couldn't we ditch the reduce and just do a bind[0] here?

File: elements/nuxeo-keys/nuxeo-keys.html (hunk 1 above)
Comment (Gabez0r) [reply elements/nuxeo-keys/nuxeo-keys.html]: Please add this information in the comment and the link to the code as well.

**Comment by Gabez0r** [comment]
👍

**Comment by guirenard** [comment]
:+1:

**Comment by guirenard** [comment]
glad to now be able to `Ctrl+c` within web ui :ok_hand:

**Pull request merged**